.github/workflows/pull-request.yml

```
@@ -253,6 +253,59 @@ jobs:
+          timeout=${HEALTH_CHECK_TIMEOUT:-120}
+          while ! nc -z localhost 4321 && [ $timeout -gt 0 ]; do
+            sleep 1
+            timeout=$((timeout-1))
+          done
+          if [ $timeout -eq 0 ]; then
+            echo "Timeout waiting for application to start"
+            exit 1
+          fi
+          curl --fail --silent http://localhost:4321/health || exit 1
+    
+      - name: Stop Docker Container 
```

yamllint

[error] trailing spaces

(trailing-spaces)